types/aws-lambda/trigger/codebuild-cloudwatch-state.d.ts

```diff
@@ -0,0 +1,110 @@
+import { Handler } from '../handler';
+
+export type CodeBuildCloudWatchStateHandler = Handler<CodeBuildCloudWatchStateEvent, void>;
+
+export type CodeBuildStateType = 'IN_PROGRESS' | 'SUCCEEDED' | 'FAILED' | 'STOPPED';
+export type CodeBuildPhaseType =
+    | 'COMPLETED'
+    | 'FINALIZING'
+    | 'UPLOAD_ARTIFACTS'
+    | 'POST_BUILD'
+    | 'BUILD'
+    | 'PRE_BUILD'
+    | 'INSTALL'
+    | 'QUEUED'
+    | 'DOWNLOAD_SOURCE'
+    | 'PROVISIONING'
+    | 'SUBMITTED';
+export type CodeBuildPhaseStatusType =
+    | 'TIMED_OUT'
+    | 'STOPPED'
+    | 'FAILED'
+    | 'SUCCEEDED'
+    | 'FAULT'
+    | 'CLIENT_ERROR';
+export type CodeBuildCacheType = 'NO_CACHE' | 'LOCAL' | 'S3';
+export type CodeBuildSourceLocationType =
+    | 'CODECOMMIT'
+    | 'CODEPIPELINE'
+    | 'GITHUB'
+    | 'GITHUB_ENTERPRISE'
+    | 'BITBUCKET'
+    | 'S3'
+    | 'NO_SOURCE';
+export type CodeBuildEnvironmentType =
+    | 'LINUX_CONTAINER'
+    | 'LINUX_GPU_CONTAINER'
+    | 'WINDOWS_CONTAINER'
+    | 'ARM_CONTAINER';
+export type CodeBuildEnvironmentPullCredentialsType = 'CODEBUILD' | 'SERVICE_ROLE';
+export type CodeBuildEnvironmentComputeType =
+    | 'BUILD_GENERAL1_SMALL'
+    | 'BUILD_GENERAL1_MEDIUM'
+    | 'BUILD_GENERAL1_LARGE'
+    | 'BUILD_GENERAL1_2XLARGE';
+export type CodeBuildEnvironmentVariableType = 'PARAMETER_STORE' | 'PLAINTEXT';
```

Comment:
I'm not a big fan of calling types "types", but really it's more important to be consistent with the property name, documentation and the rest of the package, in that order, and it's not really that important.

Reply:
I based the naming on the sdk documentation and there they explicitly use `type`
https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/CodeBuild.html